apps/beeai-sdk/examples/dependencies.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2025 © BeeAI a Series of LF Projects, LLC
+# SPDX-License-Identifier: Apache-2.0
+
+from collections.abc import AsyncGenerator
+from typing import Annotated
+
+from a2a.client import create_text_message_object
+from a2a.types import Message, TaskState, TaskStatus
+
+from beeai_sdk.a2a.extensions import LLMServiceExtensionServer, LLMServiceExtensionSpec
+from beeai_sdk.a2a.extensions.ui.trajectory import TrajectoryExtensionServer, TrajectoryExtensionSpec
+from beeai_sdk.a2a.types import RunYield
+from beeai_sdk.server import Server
+from beeai_sdk.server.context import Context
+
+server = Server()
+
+
+@server.agent()
+async def dependent_agent(
+    message: Message,
+    context: Context,
+    trajectory: Annotated[TrajectoryExtensionServer, TrajectoryExtensionSpec()],
+    # does not typecheck, does not ruff check
+    llm: LLMServiceExtensionServer = LLMServiceExtensionSpec.single_demand(),  # pyright: ignore [reportArgumentType] # noqa: B008
```

Comment:
Seeing how this does not get past linters, perhaps we want only `Annotated` for now? 😅